plugins/mmdblookup/mmdblookup.c

```diff
@@ -429,58 +475,105 @@ CODESTARTdoAction
+				value,
+				sizeof(value),
+				"%.*s",
+				entry_data.data_size,
+				entry_data.utf8_string
+			);
+			break;
+		case MMDB_DATA_TYPE_BYTES:
+			value_len = snprintf(
+				value,
+				sizeof(value),
+				"%.*s",
+				entry_data.data_size,
+				(const char *)entry_data.bytes
+			);
+			break;
+		case MMDB_DATA_TYPE_FLOAT:
+			value_len = snprintf(value, sizeof(value), "%.5f", entry_data.float_value);
+			break;
+		case MMDB_DATA_TYPE_DOUBLE:
+			value_len = snprintf(value, sizeof(value), "%.5f", entry_data.double_value);
+			break;
+		case MMDB_DATA_TYPE_UINT16:
+			value_len = snprintf(value, sizeof(value), "%" PRIu16, entry_data.uint16);
+			break;
+		case MMDB_DATA_TYPE_UINT32:
+			value_len = snprintf(value, sizeof(value), "%" PRIu32, entry_data.uint32);
+			break;
+		case MMDB_DATA_TYPE_INT32:
+			value_len = snprintf(value, sizeof(value), "%" PRIi32, entry_data.int32);
+			break;
+		case MMDB_DATA_TYPE_UINT64:
+			value_len = snprintf(value, sizeof(value), "%" PRIu64, entry_data.uint64);
+			break;
+		case MMDB_DATA_TYPE_UINT128: {
+			uint8_t *p = (uint8_t *)&entry_data.uint128;
+			value_len = snprintf(
+				value,
+				sizeof(value),
+				"0x"
+				"%02x%02x%02x%02x"
+				"%02x%02x%02x%02x"
+				"%02x%02x%02x%02x"
+				"%02x%02x%02x%02x",
+				p[0], p[1], p[2], p[3],
+				p[4], p[5], p[6], p[7],
+				p[8], p[9], p[10], p[11],
+				p[12], p[13], p[14], p[15]
+			);
+			break;
+		}
+		default:
+			dbgprintf("Got an error looking up the entry data - unknown data type\n");
+			ABORT_FINALIZE(RS_RET_OK);
+		}
 
-	size_t  memlen;
-	char   *membuf;
-	FILE   *memstream;
-	CHKmalloc(memstream = open_memstream(&membuf, &memlen));
+		if (value_len <= 0) {
+			continue;
+		}
 
-	if (entry_data_list != NULL && memstream != NULL) {
-		MMDB_dump_entry_data_list(memstream, entry_data_list, 2);
-		fflush(memstream);
-		str_split(&membuf);
-	}
+		CHKmalloc(field_json = json_object_new_string(value));
 
-	DBGPRINTF("maxmindb returns: '%s'\n", membuf);
-	total_json = json_tokener_parse(membuf);
-	fclose(memstream);
-	free(membuf);
-
-	/* extract and amend fields (to message) as configured */
-	for (int i = 0 ; i <  pData->fieldList.nmemb; ++i) {
-		char *strtok_save;
-		char buf[(strlen((char *)(pData->fieldList.name[i])))+1];
-		strcpy(buf, (char *)pData->fieldList.name[i]);
-
-		json_object *temp_json = total_json;
-		json_object *sub_obj   = temp_json;
-		const char *SEP = "!";
-
-		/* find lowest level JSON object */
-		char *s = strtok_r(buf, SEP, &strtok_save);
-		while(s != NULL) {
-			json_object_object_get_ex(temp_json, s, &sub_obj);
-			temp_json = sub_obj;
-			s = strtok_r(NULL, SEP, &strtok_save);
-		}
-		/* temp_json now contains the value we want to have, so set it */
-		json_object_get(temp_json);
-		msgAddJSON(pMsg, (uchar *)pData->fieldList.varname[i], temp_json, 0, 0);
+		// field_json now contains the geoip field, so set it to the message.
+		msgAddJSON(pMsg, (uchar *)pData->fieldList.varname[i], field_json, 0, 0);
```

Comment:
Are we leaking the `field_json` values created above, except on the last loop? Based on the `json-c` docs, I would have expected a `json_object_put` here.